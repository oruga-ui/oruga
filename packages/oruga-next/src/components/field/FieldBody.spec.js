import { shallowMount, createLocalVue } from '@vue/test-utils'
import OFieldBody from '@components/field/FieldBody'
import OField from '@components/field/Field'

const localVue = createLocalVue()
localVue.component('o-field', OField)

describe('OFieldBody', () => {
    it('is called', () => {
        const wrapper = shallowMount(OFieldBody, {
            slots: {
                default: 'content'
            }
        })
        expect(wrapper.exists()).toBeTruthy()
    })

    it('render correctly', () => {
        const wrapper = shallowMount(OFieldBody, {
            slots: {
                default: 'content'
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render the text when the child is a text', () => {
        const text = 'text'
        const wrapper = shallowMount(OFieldBody, {
            slots: {
                default: text
            }
        })
        expect(wrapper.text()).toBe(text)
    })
})
