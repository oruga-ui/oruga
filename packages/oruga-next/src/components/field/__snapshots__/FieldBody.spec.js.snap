// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OFieldBody render correctly 1`] = `<div>content</div>`;
